src/GitHub.Api/UI/TreeBase.cs

```diff
@@ -102,6 +102,28 @@ public void Load(IEnumerable<TData> treeDatas)
             }
         }
 
+        public void CheckAll(bool isChecked)
```

Comment:
Wait - this looks like it is just toggling each node, not checking each node?

Reply:
If you send `true` it will check all nodes. If you send `false` it will uncheck all nodes.
Is it the name?
Maybe the method could be called `SetCheckOnAll`?

Reply:
The name might be best changed, but I was referring to inside the foreach.

Just looking at the code walking through it I think I'm reading it right:
110: wasChecked is true if the node is checked right now and false otherwise
111: change the node...

Ah...never mind. I see it now. Sorry - this is good to go! If you want to change the name it might be clearer, but the logic is sound.